### ISSUE TITLE
Terralith's scarlet mountains turn to dirt mountains with Tectonic increased world height

### ISSUE THREAD

**Issue by johalun** [Apollounknowndev/tectonic#177]
MC 1.20.1
Forge 47.2.32
Tectonic 2.3.5a (mod)
Terralith 2.5.1 (mod)

Tectonic has a config option "increased_height" which increases the world height to 640. When setting this to true, what used to be Terralith's scarlet mountains becomes just dirt/stone mountains. The terrain itself looks very similar regardless of this setting but the blocks are changed. The biome is still scarlet mountains.

https://github.com/Stardust-Labs-MC/Terralith/issues/133

**Comment by ThariCorvy** [comment]
Happens with Volcano Peaks as well.
![image](https://github.com/Apollounknowndev/tectonic/assets/139649594/db388701-5475-4bb7-8b2a-292346814f95)

**Comment by johalun** [comment]
Is there a ETA on a potential fix? I don't mean to put stress on any devs at all, just wondering since we are planning to release a new modpack and would like to include this fix if it aligns with our planned release.

**Comment by dasaikuru** [comment]
![2024-07-07_03 58 53](https://github.com/Apollounknowndev/tectonic/assets/151808033/572a3098-8dbe-4550-911a-52f4a6367532)
happens with painted mountains as well

**Comment by KaninchenSpeed** [comment]
I found a fix, but I haven't tested if it has other implications.

Tested with:
Terratonic v3.2 ~ 1.21
Terralith 1.21_v2.5.4
on seed `-7609845261306572298`

I deleted `increased_height/data/minecraft/worldgen/noise_settings/large_biomes.json` and replaced `increased_height/data/minecraft/worldgen/noise_settings/overworld.json` with `data/minecraft/worldgen/noise_settings/overworld.json` from terralith and edited
```
"noise": {
    "height": 384,
    "min_y": -64,
    "size_horizontal": 1,
    "size_vertical": 1
}
```
to
```
"noise": {
    "height": 704,
    "min_y": -64,
    "size_horizontal": 1,
    "size_vertical": 1
}
```

**Comment by Apollounknowndev** [comment]
Fixed for a future release.

**Comment by johalun** [comment]
What version is the fix for this scheduled to be in?

**Comment by Kaidokie** [comment]
According to the changelog it's already been fixed for 1.21

**Comment by johalun** [comment]
Not 1.20.1?

**Comment by johalun** [comment]
😞  Wish it was open source so we could see the change and apply it ourselves if 1.20.1 won't be fixed...

**Comment by Apollounknowndev** [comment]
> 😞 Wish it was open source so we could see the change and apply it ourselves if 1.20.1 won't be fixed...

You're on the Github page for Tectonic???

**Comment by johalun** [comment]
Oh I see, the code is in other branches! I thought it was just for issue management since the main branch is empty. I've seen this with other mods.

**Comment by johalun** [comment]
Any chance we can get the terralith fix in 1.20 also?

**Comment by Apollounknowndev** [comment]
A backport of the fix to 1.20.1 will happen at some point but it's going to take some time as I need to make larger architectural changes to the mod on pre-1.21 versions.

**Comment by johalun** [comment]
Awesome! I appreciate your efforts.

**Comment by Apollounknowndev** [comment]
Fixed in v2.4.1 for both 1.20.1 and 1.21.
